doc/sphinx/addendum/type-classes.rst

```diff
@@ -446,26 +438,47 @@ few other commands related to typeclasses.
    + When considering local hypotheses, we use the union of all the modes
      declared in the given databases.
 
-   .. tacv:: typeclasses eauto @natural
-
-      .. warning::
-         The semantics for the limit :n:`@natural`
-         is different than for auto. By default, if no limit is given, the
-         search is unbounded. Contrary to :tacn:`auto`, introduction steps are
-         counted, which might result in larger limits being necessary when
-         searching with ``typeclasses eauto`` than with :tacn:`auto`.
+   + Note that this tactic is distinct from the similarly-named command
+     :cmd:`Typeclasses eauto`.
 
-   .. tacv:: typeclasses eauto with {+ @ident}
+   :n:`@int_or_var`
+     Specifies the maximum depth of the search.
 
-      This variant runs resolution with the given hint databases. It treats
+      .. warning::
+         The semantics for the limit :n:`@int_or_var`
+         are different than for :tacn:`auto`. By default, if no limit is given, the
+         search is unbounded. Unlike :tacn:`auto`, introduction steps count against
+         the limit, which might result in larger limits being necessary when
+         searching with :tacn:`typeclasses eauto` than with :tacn:`auto`.
+
+   :n:`with {+ @ident }`
+      Runs resolution with the specified hint databases. It treats
       typeclass subgoals the same as other subgoals (no shelving of
       non-typeclass goals in particular).
 
-.. tacn:: autoapply @term with @ident
+     When :n:`with` is specified, :tacn:`typeclasses eauto`
+     allows shelved goals to remain at any point during search and treats
+     typeclass goals like any other.
+
+     When :n:`with` is not specfied, :tacn:`typeclasses eauto` uses
+     the ``typeclass_instances`` database by default (instead of ``core``).
+     Dependent subgoals are automatically shelved, and shelved goals can
+     remain after resolution ends (following the behavior of Coq 8.5).
```

Comment:
I wonder if it wouldn't read better if the two paragraphs were inverted.

Reply:
See what I did.  Not keen on describing with, then not-with, then going back to with--think that a bit confused/confusing.

Reply:
New version looks good.